server/service/apple_mdm.go

```diff
@@ -2753,6 +2758,13 @@ func (svc *MDMAppleCheckinAndCommandService) CommandAndReportResults(r *mdm.Requ
 		if err := svc.ds.SetOrUpdateHostDisksSpace(r.Context, host.ID, availableDeviceCapacity, 100*availableDeviceCapacity/deviceCapacity, deviceCapacity); err != nil {
 			return nil, ctxerr.Wrap(r.Context, err, "failed to update host storage")
 		}
+		if err := svc.ds.UpdateHostOperatingSystem(r.Context, host.ID, fleet.OperatingSystem{
```

Comment:
Does this need to be run for every OS, or just for iOS and iPadOS?

Reply:
We need to do this here for iOS and iPadOS because they don't have osquery.
For macOS this kind of thing is run as part of the osquery query ingestion code.